cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -781,20 +794,29 @@ fun Node.followXUntilHit(
         if (nextNodes.isEmpty() && collectFailedPaths) {
             // TODO: How to determine if this path is really at the end or if it exceeded the number
             // of steps?
-            failedPaths.add(FailureReason.PATH_ENDED to currentPath.map { it.first })
+            failedPaths.add(
+                FailureReason.PATH_ENDED to
+                    NodePath(currentPath.map { it.first })
+                        .addAssumptionDependences(currentPath.map { it.second }.toList())
```

Comment:
Why this unintuitive interface of generating a `NodePath` and then calling `addAssumptionDependences` instead of directly adding them to the constructor?

Reply:
Why do we only add `it.second` as assumption dependences? Would it make sense to refactor the worklist instead and directly collect `NodePath` objects there rather than adding assumptions to a buch of different objects and copying them over and over again?